pkg/client/patch.go

```diff
@@ -84,11 +84,15 @@ type MergeFromOptions struct {
 	OptimisticLock bool
 }
 
+type patchHandler interface {
+	createPatch(originalJSON, modifiedJSON []byte, dataStruct interface{}) ([]byte, error)
+}
+
 type mergeFromPatch struct {
```

Comment:
If you want to ensure, that this doesn't break in any upcoming release, can you add a test case like the one in https://github.com/kubernetes-sigs/controller-runtime/issues/1438?